kyaml/setters2/settersutil/fieldsetter.go

```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
 	}
 
 	// Update the resources with the new value
-	inout := &kio.LocalPackageReadWriter{PackagePath: resourcesPath}
+	inout := &kio.LocalPackageReadWriter{PackagePath: resourcesPath, NoDeleteFiles: true}
 	s := &setters2.Set{Name: fs.Name}
 	err := kio.Pipeline{
 		Inputs:  []kio.Reader{inout},
-		Filters: []kio.Filter{kio.FilterAll(s)},
+		Filters: []kio.Filter{FilterAll(s)},
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 	return s.Count, err
 }
+
+// FilterAll applies the set filter for all yaml nodes and only returns the nodes whose
+// corresponding file has at least one node with input setter
+func FilterAll(s *setters2.Set) kio.Filter {
+	return kio.FilterFunc(func(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+		var setFiles sets.String = map[string]interface{}{}
+		var setNodes []*yaml.RNode
+		// for each node record the set fields count before and after filter is applied and
+		// store the corresponding file paths if there is an increment in setters count
+		for i := range nodes {
+			preCount := s.Count
+			_, err := s.Filter(nodes[i])
+			if err != nil {
+				return nil, errors.Wrap(err)
+			}
+			postCount := s.Count
```

Comment:
you don't need to save postCount, you can just do preCount < s.Count

Reply:
Yes.